core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocalEntityManager.java

```diff
@@ -475,6 +483,19 @@ private void unmanage(final Entity e, ManagementTransitionMode mode, boolean has
         entityModesById.remove(e.getId());
     }
     
+    private void pruneOrphanedLocations(Entity unmanagedEntity) {
+        Set<Location> orphanLocations = MutableSet.copyOf(unmanagedEntity.getLocations());
+        for (Entity e : getEntities()) {
+            if (e == unmanagedEntity)
+                continue;
+            orphanLocations.removeAll(e.getLocations());
+        }
+        LocationManager lm = managementContext.getLocationManager();
+        for (Location l : orphanLocations) {
+            lm.unmanage(l);
+        }
+    }
```

Comment:
Perhaps this method should be moved onto `ManagementContextInternal` and broadened to GC orphans of all types? As it stands, we're coupling `LocalEntityManger` to `(Local)LocationManager`.